api/permissions/permission_service.py

```diff
@@ -130,7 +131,16 @@ def get_permitted_environments_for_user(
     )
     filter_ = base_filter & Q(project=project)
 
-    return Environment.objects.filter(filter_).distinct().defer("description")
+    queryset = Environment.objects.filter(filter_)
+    if prefetch_metadata:
+        queryset = queryset.prefetch_related("metadata")
+
+    # Description is defered due to Oracle support where a
+    # query can't have a where clause if description is in
+    # the select parameters. This leads to an N+1 query for
+    # lists of environments when description is included, as
+    # each environment object re-queries the DB seperately.
+    return queryset.distinct().defer("description")
```

Comment:
Maybe we could branch out the logic here based on `connection.vendor`?

Reply:
I was thinking that too, though it does open up a can of worms with respect to complexity.

Reply:
Hmm... yeah, this is interesting. I guess for now, let's limit the scope of this PR to the N+1 caused by metadata. I think probably the better way to tackle this issue is to remove the `"description"` from wherever we list the environments. I've added this to the V2 API GH issue [here](https://github.com/Flagsmith/flagsmith/issues/1667) but we may be able to achieve this before then by creating a new endpoint perhaps.